packages/editor/src/client/panel/peerEditPanel.tsx

```diff
@@ -101,7 +102,7 @@ export function PeerEditPanel({id, onClose, onSave}: ImageEditPanelProps) {
           setLoadingPeerProfile(false)
 
           // TODO: Better validation
-          if (response?.data?.peerProfile) {
+          if (response?.data?.peerProfile && response?.data?.peerProfile.hostURL !== hostURL) {
```

Comment:
I would make an extra check for this so you can also set a specific error "You can not peer with yourself".

Reply:
Can I perform the check inside the form control since it's just a boolean here?